### PR TITLE
feat(ffi): logging support for android

### PR DESCRIPTION
**Pull request by gnunicorn**
And introduce the concept of platform-specific implementations via a `platforms.rs` mod. Probably to be moved into a subfolder once this becomes too much...

### CODE REVIEW

File: bindings/matrix-sdk-ffi/Cargo.toml [hunk 1]
```diff
@@ -27,14 +27,23 @@ futures-signals = { version = "0.3.30", default-features = false }
...
-tracing-subscriber = { version = "0.3", features = ["env-filter"] }
 uniffi = { workspace = true }
 uniffi_macros = { workspace = true }
+
+
+[target.'cfg(target_os = "android")'.dependencies]
+tracing = { version = "0.1.29", default-features = false, features = ["log"] }
+android_logger = "0.11"
+log-panics = { version = "2", features = ["with-backtrace"]}
+matrix-sdk = { path = "../../crates/matrix-sdk", default-features = false, features = ["anyhow", "experimental-timeline", "e2e-encryption", "sled", "markdown", "sliding-sync", "socks", "rustls-tls"], version = "0.6.0" }
```
Comment (stefanceriu) [inline bindings/matrix-sdk-ffi/Cargo.toml:44]: Hmm.. shouldn't iOS also have `e2e-encryption` and `sled`? 🤔

File: bindings/matrix-sdk-ffi/src/platform.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+mod other {
+    use std::io;
+
+    use tracing_subscriber::{fmt, prelude::*, EnvFilter};
+
+    #[uniffi::export]
+    fn setup_tracing(configuration: String) {
+        tracing_subscriber::registry()
+            .with(EnvFilter::new(configuration))
+            .with(fmt::layer().with_ansi(false).with_writer(io::stderr))
```
Comment (poljar) [inline bindings/matrix-sdk-ffi/src/platform.rs:50]: If we're going to have an other here we might want to enable colors. AFAIR they are problematic for iOS while they do work on Android and probably for other platforms as well.

**Review by poljar — APPROVED** [review]

**Review by stefanceriu — APPROVED** [review]
This looks great, really useful 👏

File: bindings/matrix-sdk-ffi/Cargo.toml (hunk 1 above)
Comment (ganfra) [reply bindings/matrix-sdk-ffi/Cargo.toml:44]: It's enabled by default, so you don't have to, it's just that for android we have to use rustls-tls and disable default

Comment (stefanceriu) [reply bindings/matrix-sdk-ffi/Cargo.toml:44]: Right, got it! 👍

File: bindings/matrix-sdk-ffi/src/platform.rs (hunk 2 above)
Comment (gnunicorn) [reply bindings/matrix-sdk-ffi/src/platform.rs:46]: I just used the existing as a fallback to not have compile errors for the host-platform. Sure we can improve that if you want. just a `with_ansi(true)` then?

Comment (gnunicorn) [reply bindings/matrix-sdk-ffi/src/platform.rs:46]:
```suggestion
            .with(fmt::layer().with_ansi(true).with_writer(io::stderr))
```

Comment (poljar) [reply bindings/matrix-sdk-ffi/src/platform.rs]: > I just used the existing as a fallback to not have compile errors for the host-platform. Sure we can improve that if you want. just a `with_ansi(true)` then?

Yep.

**Pull request merged**